### PR TITLE
55 resolve product type key

### PR DESCRIPTION
**Pull request by lojzatran**
Reopen the PR https://github.com/commercetools/commercetools-node-variant-reassignment/pull/56. I thought it's already covered in master, but it's not, so I reopen the issue.

### CODE REVIEW

File: lib/runner/variant-reassignment.js [hunk 1]
```diff
@@ -266,6 +269,20 @@ export default class VariantReassignment {
...
   }
 
+  /**
+   * In productDrafts from external systems, there's no productTypeId, but instead the name is used.
+   * However, productTypeId is needed for reassignment. This method replaces the productTypeName
+   * with productTypeId if such ID exists.
+   */
+  _resolveProductTypeReferences (productDrafts, productTypeIdToTypeObj) {
+    productDrafts.forEach((productDraft) => {
+      const productType = productTypeIdToTypeObj[productDraft.productType.id]
```
Comment (junajan) [inline lib/runner/variant-reassignment.js]: If the `productDraft.productType.id` variable contains productType name we should rename `productTypeIdToTypeObj` to something like `productTypeNameToTypeObj`.

File: spec/unit/runner/variant-reassignment.spec.js [hunk 2]
```diff
@@ -367,4 +367,32 @@ describe('Variant reassignment', () => {
...
+      sinon.stub(productServiceMock, 'fetchProductsFromProductDrafts')
+        .resolves(null)
+      const variantReassignments = new VariantReassignment(null, logger)
+      sinon.stub(variantReassignments, '_processUnfinishedTransactions')
+        .resolves(null)
+      const testFunction = sinon.stub(variantReassignments, '_selectProductDraftsForReassignment')
+        .returns([])
+      variantReassignments.productService = productServiceMock
+
+      const productTypeKey = 'productTypeKey'
```
Comment (junajan) [inline spec/unit/runner/variant-reassignment.spec.js]: We resolve productType reference using `productType.name` but here we use `productTypeKey`

File: lib/runner/variant-reassignment.js (hunk 1 above)
Comment (lojzatran) [reply lib/runner/variant-reassignment.js]: https://github.com/commercetools/commercetools-node-variant-reassignment/pull/57/commits/51c26c3a1605a98f7959435daec1a36e470e745d

File: spec/unit/runner/variant-reassignment.spec.js (hunk 2 above)
Comment (lojzatran) [reply spec/unit/runner/variant-reassignment.spec.js]: https://github.com/commercetools/commercetools-node-variant-reassignment/pull/57/commits/3d17ad118cdc9d4aad943f529101d52d6f3d57d8

**Pull request merged**